packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +36,42 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold gemoetry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.bottomNavigationBar].
+  /// Null when there is no bottom navigation bar..
```

Comment:
nit: these docs need work ("Null" should be lowercase, since Null is the type and null is the value; double period at the end; blank line should be between the two sentences, needs more detail)

Reply:
change the nits, though I'm not sure which more details are missing here.
(though we might be changing this to not be Rect at all, so maybe not relevant)